File: test/four/app/src/main/java/com/z3d/material/DynamicBaseConstItem.java
package com.z3d.material;

import java.util.ArrayList;
import java.util.List;

public class DynamicBaseConstItem {
    private static final String TAG ="DynamicBaseConstItem" ;
    public ConstItem target;
    public String paramName ;
    public List<Float> currentValue;
    public int targetOffset;
    protected int _type;
    public void setTargetInfo(ConstItem $target, String $paramName, int $type) {
        this.target = $target;
        this.paramName = $paramName;
        this.settype($type);
        if( this.target!=null){
            this.target.setDynamicOffset(this);
        }

        this.currentValue = new ArrayList<>($type);

    }
    public void settype(int value) {
        this._type = value;
    }
    public void setCurrentVal(List args ) {
        for (int i = 0; i < args.size(); i++) {
            this.currentValue.add(i,(float)args.get(i));
        }
    }

    public void update() {
        if (this.target!=null) {

            this.target.setDynamic(this);
        }
    }
}

File: metalios/iosgl/iosgl/pan3d/material/DynamicBaseConstItem.m
//
//  DynamicBaseConstItem.m
//  iosgl
//
//  Created by zhao on 9/3/2020.
//  Copyright © 2020 zhao. All rights reserved.
//

#import "DynamicBaseConstItem.h"
#import "ConstItem.h"
 
@implementation DynamicBaseConstItem
 
-(void)update:(float)t;
{
    if(self.target){
        [self.target setDynamic:self];
    }
    
}

-(void)setTargetInfo:(ConstItem*)target paramName:(NSString*)paramName type:(int)type;
{
    self.target = target;
    self.paramName = paramName;
    self.type = type;
    [self.target setDynamicOffset: self];
    self.currentValue=[[NSMutableArray alloc]init];
    for(int i=0;i<type;i++){
        [self.currentValue addObject:@0];
    }
    
}
-(void)setType:(int)value;
{
  self._type=value;
}
-(int)type;
{
    return self._type;
}
-(void)setCurrentVal:(NSNumber*)x;
{
}
-(void)setCurrentVal:(NSNumber*)x y:(NSNumber*)y;
 {
 }
-(void)setCurrentVal:(NSNumber*)x y:(NSNumber*)y z:(NSNumber*)z;
{
}
@end
